build.zig

```diff
@@ -40,6 +40,11 @@ pub fn build(b: *std.Build) !void {
         "-fno-sanitize=undefined",
     };
 
+    const openssl = switch (target.result.os.tag) {
+        .linux => b.option(bool, "enable-openssl", "Enable OpenSSL support") orelse false,
+        else => false,
+    };
```

Comment:
This option should be visible regardless of the compile target, with the description mentioning Linux only. This gives users running `zig build -h` a better idea of what's supported regardless of their host target.

```suggestion
    const openssl = (b.option(bool, "enable-openssl", "Use OpenSSL instead of MbedTLS (Linux only)") orelse false) and target.result.os.tag == .linux;
```

Reply:
Maybe let users decide on this option. Because, i am pretty sure openssl can be build on BSD and macos too.
```suggestion
    const openssl = switch (target.result.os.tag) {
        .windows => false,
        else => b.option(bool, "enable-openssl", "Enable OpenSSL support") orelse false,
    };
```

Reply:
Sure.
```suggestion
    const openssl = (b.option(bool, "enable-openssl", "Enable OpenSSL support") orelse false) and target.result.os.tag != .windows;
```